service/domain/blobs/replication/manager.go

```diff
@@ -2,118 +2,78 @@ package replication
 
 import (
 	"context"
-	"time"
+	"sync"
 
 	"github.com/boreq/errors"
 	"github.com/planetary-social/go-ssb/logging"
-	"github.com/planetary-social/go-ssb/service/domain/blobs"
 	"github.com/planetary-social/go-ssb/service/domain/messages"
-	"github.com/planetary-social/go-ssb/service/domain/refs"
 	"github.com/planetary-social/go-ssb/service/domain/transport"
 	"github.com/planetary-social/go-ssb/service/domain/transport/rpc"
 )
 
-type WantListStorage interface {
-	GetWantList() (blobs.WantList, error)
-}
-
-type Downloader interface {
-	OnHasReceived(ctx context.Context, peer transport.Peer, blob refs.Blob, size blobs.Size)
-}
-
 type Manager struct {
-	storage    WantListStorage
-	downloader Downloader
-	logger     logging.Logger
+	wantListStorage WantListStorage
+	blobStorage     BlobSizeRepository
+	downloader      Downloader
+	logger          logging.Logger
+
+	// todo cleanup processes
```

Comment:
I think this can be done later, it could even stay like this all the way to production unless we ran the app for a very long time.

Reply:
As I understand the underlying process goroutines for incoming and outgoing loops will get terminated on connection context termination anyway, right? So the only thing we'd have to improve would be reacting to connection termination by deleting the process struct from this map (for example by running a goroutine awaiting ctx close while creating a new process in `getOrCreateProcess`). Do I understand the problem correctly?
I'm actually ok with leaving it as it is if that's only about cleaning up the process struct. As I can imagine the map won't grow huge in our normal use case.

Reply:
Correct.